### DISCUSSION TITLE
Persistence / Retries for background tasks

### DISCUSSION THREAD

**Discussion by nileshtrivedi** [encode/starlette#2628] [Ideas] [answered]
This is a speculative question. I came to know about [DEP 0014](https://github.com/django/deps/blob/main/accepted/0014-background-workers.rst) (Django Enhancement Proposal) - which plans to standardize and build support for background tasks within Django itself (using asyncio).

Starlette already supports [Background Tasks](https://www.starlette.io/background/). Would it be too difficult to add persistence for these background jobs, say with PostgreSQL's `SKIP LOCKED` queries or [ValKey](https://github.com/valkey-io/valkey) ? 🤔 

I have a FastAPI + PostgreSQL app in production that needs Celery and Redis for background jobs. If Starlette (and therefore FastAPI) supported persisted background jobs with Postgres as backend, it would simplify my infra a lot.

**Comment by adriangb** [comment, marked as answer]
You can easily write your own background task system, it is loosely coupled. See for example https://github.com/adriangb/asgi-background.

**Reply by nileshtrivedi** [reply]
Thanks for the pointer. It's helpful. 👍 

Saw this on your asgi-background repo:

> Unlike Starlette, we do not execute background tasks within the ASGI request/response cycle. Instead we schedule them in a TaskGroup that is bound to the application's lifespan

This seems quite useful as a built-in feature for Starlette. Has it been considered/evaluated for merging?

**Reply by adriangb** [reply]
There's no plan to incorporate it into Starlette.
